Source/missiles.h

```diff
@@ -153,6 +153,11 @@ struct Missile {
 		return &Monsters[_misource];
 	}
 
+	[[nodiscard]] bool isSameSource(Missile &missile)
+	{
+		return _misource == missile._misource;

```

Comment:
out of context of the specific spell this would need to also check the type of caster or monsters and players could collide ... of cause golems do not cast spells or shoot bows.

Reply:
I added a check for the same sourceType too, do you think that's enough?